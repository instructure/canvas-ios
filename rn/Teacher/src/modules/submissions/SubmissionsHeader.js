//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import React, { Component } from 'react'
import {
  View,
} from 'react-native'

import i18n from 'format-message'
import { LinkButton } from '../../common/buttons'
import { Heading1, Text } from '../../common/text'
import { createStyleSheet } from '../../common/stylesheet'
import { joinTitles } from '../filter/filter-options'

export type SubmissionsHeaderProps = {
  filterOptions: SubmissionFilterOption[],
  selectedFilter?: ?SelectedSubmissionFilter,
  onSelectFilter: Function,
  navigator: Navigator,
}

export default class SubmissionsHeader extends Component<SubmissionsHeaderProps, any> {
  navigateToFilter = () => {
    this.props.navigator.show('/filter', {
      modal: true,
    }, { ...this.props })
  }

  render () {
    let title = joinTitles(this.props.filterOptions) || i18n('All Submissions')

    let subTitle = ''
    if (this.props.anonymous) {
      subTitle = i18n('Anonymous grading')
    }

    return (<View style={styles.headerWrapper}>
      <View style={styles.header}>
        <Heading1 numberOfLines={1}>
          { title }
        </Heading1>
        {!!subTitle && <Text testID='SubmissionsHeader.subtitle' style={styles.subtitle}>{subTitle}</Text>}
      </View>
      { this.renderFilterButton() }
    </View>)
  }

  renderFilterButton = () => {
    let title = i18n('Filter')
    let onPress = this.navigateToFilter

    let selected = this.props.filterOptions.filter(option => option.selected)
    if (selected.length > 0) {
      title = i18n('Filter ({numSelected, number})', { numSelected: selected.length })
    }

    let accessibilityLabel = i18n('Filter Submissions. ({numSelected, number}) selected.', { numSelected: selected.length })

    return (<LinkButton
      testID='submission-list.filter'
      onPress={onPress}
      style={styles.filterButton}
      accessibilityLabel={ accessibilityLabel }
    >
      { title }
    </LinkButton>)
  }
}

const styles = createStyleSheet((colors, vars) => ({
  headerWrapper: {
    borderBottomWidth: vars.hairlineWidth,
    borderBottomColor: colors.borderMedium,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingTop: 16,
    paddingBottom: 8,
    paddingHorizontal: 16,
    height: 'auto',
  },
  header: {
    flex: 1,
    flexDirection: 'column',
  },
  subtitle: {
    color: colors.textDark,
    fontSize: 14,
    fontWeight: '500',
  },
  filterButton: {
    marginBottom: 1,
    width: 'auto',
  },
}))
