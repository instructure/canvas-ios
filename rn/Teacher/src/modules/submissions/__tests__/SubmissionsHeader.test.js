//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
import { shallow } from 'enzyme'
import React from 'react'
import SubmissionsHeader from '../SubmissionsHeader.js'
import defaultFilterOptions from '../../filter/filter-options'

let template = {
  ...require('../../../__templates__/helm'),
}

jest.mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity')

describe('SubmissionHeader', () => {
  it('navigates to filter when pressed', () => {
    let navigator = template.navigator()
    let filterOptions = defaultFilterOptions()
    const tree = shallow(
      <SubmissionsHeader filterOptions={filterOptions} navigator={navigator} />
    )

    let filterButton = tree.find('[testID="submission-list.filter"]')
    filterButton.simulate('Press')

    expect(navigator.show).toHaveBeenCalledWith(
      '/filter',
      { modal: true },
      { filterOptions, navigator }
    )
  })

  it('renders anonymous grading', () => {
    const tree = shallow(
      <SubmissionsHeader filterOptions={defaultFilterOptions()} anonymous />
    )
    expect(tree.find('[testID="SubmissionsHeader.subtitle"]').prop('children')).toBe('Anonymous grading')
  })
})

