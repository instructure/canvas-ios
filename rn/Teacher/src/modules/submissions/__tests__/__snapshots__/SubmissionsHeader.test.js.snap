// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionHeader anonymous and muted 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#C7CDD1",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 24,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
        ]
      }
    >
      Grades muted, Anonymous grading
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions. (0) selected."
    accessibilityRole="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#008EE2",
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader anonymous grading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#C7CDD1",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 24,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
        ]
      }
    >
      Anonymous grading
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions. (0) selected."
    accessibilityRole="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#008EE2",
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader muted grading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#C7CDD1",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 24,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
        ]
      }
    >
      Grades muted
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions. (0) selected."
    accessibilityRole="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#008EE2",
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader muted grading but new gradebook 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#C7CDD1",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 24,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      All Submissions
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions. (0) selected."
    accessibilityRole="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#008EE2",
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SubmissionHeader muted grading but new gradebook 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#C7CDD1",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessibilityRole="header"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 24,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      All Submissions
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Object {
            "color": "#8B969E",
            "fontSize": 14,
            "fontWeight": "500",
          },
        ]
      }
    >
      Anonymous grading
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="Filter Submissions. (0) selected."
    accessibilityRole="button"
    hitSlop={
      Object {
        "bottom": 15,
        "left": 15,
        "right": 15,
        "top": 15,
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 1,
        "width": "auto",
      }
    }
    testID="submission-list.filter"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#008EE2",
                "fontSize": 16,
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Filter
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
